app_dart/lib/src/service/dynamic_config_updater.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:cocoon_server/logging.dart';
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:retry/retry.dart';
+import 'package:yaml/yaml.dart' show YamlMap, loadYaml;
+
+import '../foundation/utils.dart';
+import 'config.dart';
+
+class DynamicConfigUpdater {
+  DynamicConfigUpdater({
+    this.delay = const Duration(minutes: 1),
+    @visibleForTesting Random? random,
+    @visibleForTesting Client? httpClient,
+    @visibleForTesting
+    RetryOptions retryOptions = const RetryOptions(
+      maxAttempts: 3,
+      delayFactor: Duration(seconds: 3),
+    ),
+  }) : _random = random ?? Random(),
+       _httpClient = httpClient ?? Client(),
+       _retryOptions = retryOptions;
+
+  final Duration delay;
+  final Random _random;
+  final Client _httpClient;
+  final RetryOptions _retryOptions;
+
+  /// Fetches and parses the `config.yaml` from HEAD `flutter/cocoon/app_dart/`.
+  Future<DynamicConfig> fetchDynamicConfig() async {
+    final file = await githubFileContent(
+      Config.cocoonSlug,
+      'app_dart/config.yaml',
+      httpClientProvider: () => _httpClient,
+      retryOptions: _retryOptions,
+    );
+    final configYaml = loadYaml(file) as YamlMap;
```

Comment:
I think you can do `as YamlMap<String, Object?>` (instead of below)

Reply:
I have to cast it this way. `YamlMap` doesn't have generic types.